src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -1439,7 +1439,7 @@ void network::transfer_memory_to_device(std::shared_ptr<primitive_inst> instance
     if (!get_engine().supports_allocation(allocation_type::usm_device))
         return;
 
-    if (get_engine().get_device_info().dev_type != device_type::discrete_gpu)
+    if (!get_engine().get_device_info().has_separate_cache)
```

Comment:
` || node.is_shape_infer_dep()`

Reply:
This function is called only for constant node and it is checked in `network::allocate_primitive_instance`. Is it still necessary to check `is_shape_infer_dep`?

Reply:
There are still constant nodes being used as shape infer dep . 
E.g., strided_slice ..which should be handled as attributes though, just not applied exhaustively. 
Also, "shape_infer_dep" means that the constant will be read by host at user nodes' calc_output_layout. 
So "if there is such a case" it is better not to be transfered to device because read usm_device by host requires memcpy.

Reply:
oh I see. Let me check this and if there is any affected layers, we need to check performance impact again..